core/src/solution_submission.rs

```diff
@@ -186,6 +186,12 @@ async fn retry_with_gas_price_increase(
     const BLOCK_TIMEOUT: usize = 2;
     const DEFAULT_GAS_PRICE: u64 = 15_000_000_000;
 
+    // openethereum requires that the gas price of the resubmitted transaction has increased by at
+    // least 12.5%.
+    let min_gas_price_increase_factor: f64 = 1.125 * (1.0 + f64::EPSILON);
```

Comment:
Epsilon is always > 0 afaik since it is related to the error floating point operations have.

Reply:
`EPSILON` is a Rust standard library constant that is:
> This is the difference between 1.0 and the next larger representable number.

So basically, because of how double precision floating point numbers work, there are no values that can be represented between `1.125` and `1.125 * (1.0 + f64::EPSILON) == 1.125_000_000_000_000_2`. If you try to write express any number between those two, it will be either rounded up or down. Specifically:
```
println!("{}", 1.125_000_000_000_000_05 == 1.125);                          // true
println!("{}", 1.125_000_000_000_000_1 == 1.125);                           // true
println!("{}", 1.125_000_000_000_000_15 == (1.125 * (1.0 + f64::EPSILON))); // true
```